src/librustc/hir/mod.rs

```diff
@@ -934,7 +934,8 @@ pub struct BodyId {
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
 pub struct Body {
     pub arguments: HirVec<Arg>,
-    pub value: Expr
+    pub value: Expr,
+    pub is_generator: bool,
```

Comment:
I think @alexcrichton created `IsGenerator` because it's more clear than a `bool` in the `ExprClosure` variant.